cypress/pages/BounceRulesContainer/index.js

```diff
@@ -82,42 +79,82 @@ class BounceRulesPage extends Page {
     return cy.get(Selectors.confirmationSubmit);
   }
 
-  get testRule() {
-    return cy.get("[data-rule='cypressTest']");
+  get createdDeleteRuleAPI() {
+    return cy.get('[data-cypress="cypressDeleteTest"]');
   }
 
-  fillForm(
-    priority,
-    bounceAction,
-    responseCode,
-    description,
-    enhancedCode,
-    regex
-  ) {
-    if (priority) {
-      this.priority.type(priority);
-    }
-
-    if (bounceAction) {
-      this.bounceAction.type(bounceAction);
-    }
-
-    if (responseCode) {
-      this.responseCode.type(responseCode);
-    }
-
-    if (description) {
-      this.description.type(description);
-    }
-    if (enhancedCode) {
-      this.enhancedCode.type(enhancedCode);
-    }
+  get createdCreateRuleAPI() {
+    return cy.get('[data-cypress="cypressCreateTest"]');
+  }
 
-    if (regex) {
-      this.regex.type(regex);
-    }
+  open() {
+    super.open("/bounce_rules");
+  }
 
-    return this.submitButton.click();
+  createdRuleButton(id) {
+    return cy.get(`[data-rule="${id}"]`);
+  }
+
+  deleteBounceRuleAPI(testRule) {
+    return new Promise(resolve => {
+      cy.task("getRules").then(res => {
+        if (!res.errno) {
+          for (let i = 0; i < res.length; i++) {
+            if (_.isEqual(testRule, _.omit(res[i], "id"))) {
+              cy.task("deleteRule", res[i].id);
```

Comment:
It's bit tough to grasp the chaining of things to be less nested and behaving the way you expect but here is one way to possibly reorganize this
```
return cy.task("getRules").then(res => {
// do some filtering and then if possible
return cy.task("deleteRule")
// if not possible return a truthy value so the chain of the function will check true or false for success/failure in deleting
})
```
and in the tasks we often return back a true/false given success/error of the API call  to indicate whether or not to log success/error

Reply:
was wondering if you can give an example of this, still a bit confused by this. thanks!